src/main/java/com/internet/shop/dao/impl/ShoppingCartDaoImpl.java

```diff
@@ -0,0 +1,43 @@
+package com.internet.shop.dao.impl;
+
+import com.internet.shop.dao.ShoppingCartDao;
+import com.internet.shop.db.Storage;
+import com.internet.shop.lib.Dao;
+import com.internet.shop.model.ShoppingCart;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.IntStream;
+
+@Dao
+public class ShoppingCartDaoImpl implements ShoppingCartDao {
+    @Override
+    public ShoppingCart create(ShoppingCart shoppingCart) {
+        Storage.addShoppingCart(shoppingCart);
+        return shoppingCart;
+    }
+
+    @Override
+    public Optional<ShoppingCart> getByUserId(Long id) {
+        return Storage.carts.stream()
+                .filter(cart -> cart.getUserId().equals(id))
+                .findFirst();
+    }
+
+    @Override
+    public List<ShoppingCart> getAll() {
+        return Storage.carts;
+    }
+
+    @Override
+    public ShoppingCart update(ShoppingCart shoppingCart) {
+        IntStream.range(0, Storage.carts.size())
+                .filter(index -> Storage.carts.get(index).getId().equals(shoppingCart.getId()))
+                .forEach(item -> Storage.carts.set(item, shoppingCart));
```

Comment:
```suggestion
                .forEach(i/index -> Storage.carts.set(i/index, shoppingCart));
```
it's not item it's index

Reply:
done